java/parser.js

```diff
@@ -1,171 +1,206 @@
+        return string;
     }
 
     function accessibilityRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.accessibility(${accessibility(val)})`, ``)
+        return arr.reduce((acc, val) => `${acc}.accessibility(${accessibility(val)})`, ``);
     }
 
     function accessibility(val) {
-        return `${regionParameter(val.region)}, AccessibilityRegionType.${capitalizeFirstLetter(val.type)}`
+        return `${regionParameter(val.region)}, AccessibilityRegionType.${capitalizeFirstLetter(val.type)}`;
     }
 
     function regionParameter(region) {
-        let string
+        let string;
         switch (typeof region) {
             case 'string':
-                string = `By.cssSelector(${JSON.stringify(region)})`
+                string = `By.cssSelector(${JSON.stringify(region)})`;
                 break;
             case "object":
                 if (region.type) {
-                    string = native ? `getDriver().findElement(${parseObject(region)})` : parseObject(region)
+                    string = native ? `getDriver().findElement(${parseObject(region)})` : parseObject(region);
                 } else {
-                    string = parseObject({value: region, type: 'Region'})
+                    string = parseObject({value: region, type: 'Region'});
                 }
                 break;
             case "function":
-                string = serialize(region)
+                string = serialize(region);
                 break;
             default:
-                throw new Error(`Region parameter of the unimplemented type was used:  ${JSON.stringify(region)}`)
+                throw new Error(`Region parameter of the unimplemented type was used:  ${JSON.stringify(region)}`);
         }
         return string
     }
+
+    function checkOptions(actual, supported) {
+        const actualOptions = Object.keys(actual);
+        actualOptions.forEach(option => {
+            if (!supported.includes(option)) throw new Error(`Emitter need update to support check settings option: ${option}`);
+        })
+    }
+
 }
 
 // General
 
 function java(chunks, ...values) {
-    const commands = []
-    let code = ''
+    const commands = [];
+    let code = '';
     values.forEach((value, index) => {
         if (typeof value === 'function' && !value.isRef) {
-            code += chunks[index]
-            commands.push(code, value)
-            code = ''
+            code += chunks[index];
+            commands.push(code, value);
+            code = '';
         } else {
-            code += chunks[index] + serialize(value)
+            code += chunks[index] + serialize(value);
         }
     })
-    code += chunks[chunks.length - 1]
-    commands.push(code)
-    return commands
+    code += chunks[chunks.length - 1];
+    commands.push(code);
+    return commands;
 }
 
 function serialize(value) {
-    let stringified = ''
+    let stringified = '';
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
    let stringified = "";
```